### PR TITLE
feat(builder): get the slug/docker builder images from the environment

### LINKED ISSUE #66: Make slugbuilder and dockerimage images configurable
Adapted from deis/workflow#135

The image used for building slugs should be configurable via an environment variable (e.g. `SLUGBUILDER_IMAGE_NAME`) so that a cluster administrator can specify their own image to use. If the environment variable is missing, builder should default to the `quay.io/deisci/slugbuilder` image.

The same requirement will likely apply to the dockerbuilder image, but that's out of scope of this issue.

If so, we have to consider if someone specifies a slugbuilder or dockerbuilder image that's on the internal registry. For example, what happens when someone specifies an image that lives on the internal docker registry, the `docker pull` is in progress, and the registry crashes?

### PR DESCRIPTION
**Pull request by kmala**
fixes #66 

*Update by mboersma*:
Closes deis/workflow#135.
Closes deis/workflow#323.

### CODE REVIEW

File: pkg/gitreceive/k8s_util.go [hunk 1]
```diff
@@ -25,6 +24,11 @@ const (
 	dockerSocketPath = "/var/run/docker.sock"
 )
 
+var (
+	slugBuilderImage   = getenv("SLUGBUILDER_IMAGE_NAME", "quay.io/deisci/slugbuilder:v2-beta")
+	dockerBuilderImage = getenv("DOCKERBUILDER_IMAGE_NAME", "quay.io/deisci/dockerbuilder:v2-beta")
```
Comment (arschles) [inline pkg/gitreceive/k8s_util.go]: @kmala can you use the [envconfig](https://github.com/kelseyhightower/envconfig) [struct](https://github.com/deis/builder/blob/master/pkg/gitreceive/config.go#L13-L34) that we have already to get these environment variables?

Comment (arschles) [inline pkg/gitreceive/k8s_util.go]: also, the change I mentioned above will require that the values get passed down into the `dockerBuilderPod` and `slugBuilderPod` `func`s, so can you add to appropriate tests to ensure that the correct images are set?

Comment (kmala) [inline pkg/gitreceive/k8s_util.go]: @arschles can you check now again?

File: pkg/gitreceive/k8s_util_test.go [hunk 2]
```diff
@@ -53,18 +55,18 @@ func TestBuildPod(t *testing.T) {
...
-		{true, true, "test", "default", env, "tar", "put-url", "buildpack"},
-		{true, false, "test", "default", env, "tar", "put-url", "buildpack"},
+		{true, true, "test", "default", emptyEnv, "tar", "put-url", "", ""},
+		{true, false, "test", "default", emptyEnv, "tar", "put-url", "", ""},
+		{true, true, "test", "default", env, "tar", "put-url", "", ""},
+		{true, false, "test", "default", env, "tar", "put-url", "", ""},
+		{true, true, "test", "default", emptyEnv, "tar", "put-url", "buildpack", ""},
+		{true, false, "test", "default", emptyEnv, "tar", "put-url", "buildpack", ""},
+		{true, true, "test", "default", env, "tar", "put-url", "buildpack", ""},
+		{true, false, "test", "default", env, "tar", "put-url", "buildpack", ""},
```
Comment (arschles) [inline pkg/gitreceive/k8s_util_test.go]: could you add a row here to test for a custom image name? same with the below table for docker builds

File: pkg/gitreceive/k8s_util.go (hunk 1 above)
Comment (arschles) [inline pkg/gitreceive/k8s_util.go]: @kmala just went over it. only https://github.com/deis/builder/pull/243#discussion_r55615726 remains. thanks for making the changes

File: pkg/gitreceive/k8s_util_test.go (hunk 2 above)
Comment (kmala) [inline pkg/gitreceive/k8s_util_test.go]: added tests

**Comment by arschles** [comment]
@kmala code LGTM. I haven't run any manual tests though

**Pull request merged**